### PR TITLE
Bind and configure AuthServiceProviderRegistry to new gateway

### PR DESCRIPTION
**Pull request by seehamrun**
WIP: This does not dynamically populate the registry yet. Need to wire up a few more things before we have something runnable to check against. Also only one AuthProvider is currently set up.

### CODE REVIEW

**Comment by seehamrun** [comment]
I made the changes we discussed offline: 
-> only binding services that are enabled (as determined by flags in main or something else)
-> moving installation of auth registry module to the main class and not as part of the api module. 

PTAL

File: portability-gateway/src/main/java/org/dataportabilityproject/gateway/PortabilityAuthServiceProviderModule.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import org.slf4j.LoggerFactory;
+
+public class PortabilityAuthServiceProviderModule extends AbstractModule {
+
+  private static final Logger logger =
+      LoggerFactory.getLogger(PortabilityAuthServiceProviderModule.class);
+
+  private final ImmutableList<String> enabledServices;
+
+  PortabilityAuthServiceProviderModule(List<String> enabledServices) {
```
Comment (holachuy) [inline portability-gateway/src/main/java/org/dataportabilityproject/gateway/PortabilityAuthServiceProviderModule.java]: Maybe this should be a Set unless we care about order.

File: portability-gateway/src/main/java/org/dataportabilityproject/gateway/PortabilityAuthServiceProviderModule.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import org.dataportabilityproject.spi.gateway.auth.AuthServiceProviderRegistry;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class PortabilityAuthServiceProviderModule extends AbstractModule {
+
+  private static final Logger logger =
+      LoggerFactory.getLogger(PortabilityAuthServiceProviderModule.class);
+
+  private final ImmutableList<String> enabledServices;
```
Comment (holachuy) [inline portability-gateway/src/main/java/org/dataportabilityproject/gateway/PortabilityAuthServiceProviderModule.java]: ImmutableSet ?

File: portability-gateway/src/main/java/org/dataportabilityproject/gateway/PortabilityAuthServiceProviderRegistry.java [hunk 3]
```diff
@@ -33,34 +34,29 @@
...
-            type,
-            service);
-        supportedImportTypesBuilder.add(type);
-      }
-
-      supportedExportTypesBuilder.addAll(exportTypes);
-      serviceProviderBuilder.put(service, authServiceProvider);
-    }
+    serviceProviderMap.forEach(
+        (service, provider) -> {
```
Comment (holachuy) [inline portability-gateway/src/main/java/org/dataportabilityproject/gateway/PortabilityAuthServiceProviderRegistry.java]: fanceee!!!!

File: portability-gateway/src/main/java/org/dataportabilityproject/gateway/PortabilityAuthServiceProviderModule.java (hunk 2 above)
Comment (seehamrun) [reply portability-gateway/src/main/java/org/dataportabilityproject/gateway/PortabilityAuthServiceProviderModule.java]: Done

File: portability-gateway/src/main/java/org/dataportabilityproject/gateway/PortabilityAuthServiceProviderModule.java (hunk 1 above)
Comment (seehamrun) [reply portability-gateway/src/main/java/org/dataportabilityproject/gateway/PortabilityAuthServiceProviderModule.java]: Done

**Pull request merged**